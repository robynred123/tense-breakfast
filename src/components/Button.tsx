import { Button } from '@mui/material';
import { GREY, LIGHT_BLUE, TEAL, WHITE } from '../constants/colours';

type ButtonProps = {
  text: string;
  onClick: () => void;
  buttonColour: 'green' | 'grey' | 'gradient';
  disabled: boolean;
  width: string;
};

const setStyle = (buttonColour: string, sharedStyling: {}) => {
  switch (buttonColour) {
    case 'green':
      return {
        ...sharedStyling,
        backgroundColor: TEAL,
      };
    case 'grey':
      return {
        ...sharedStyling,
        color: WHITE,
        backgroundColor: GREY,
      };
    case 'gradient':
      return {
        ...sharedStyling,
        color: 'black',
        backgroundImage: `linear-gradient(140deg, ${TEAL}, ${LIGHT_BLUE})`,
      };
    default:
      return {
        backgroundColor: TEAL,
      };
  }
};

export const ButtonComponent = (props: ButtonProps) => {
  const { onClick, text, buttonColour, disabled, width } = props;

  const sharedStyling = {
    width: width,
    borderRadius: '15px',
  };

  return (
    <Button sx={setStyle(buttonColour, sharedStyling)} onClick={() => onClick} disabled={disabled}>
      {text}
    </Button>
  );
};
