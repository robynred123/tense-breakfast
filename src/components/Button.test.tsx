/* eslint-disable testing-library/await-async-query */
import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonComponent, ButtonProps } from './Button';

describe('<ButtonComponent />', () => {
  it('Should render the button component successfully', () => {
    const props: ButtonProps = {
      text: 'Button!',
      onClick: () => jest.fn(),
      buttonColour: 'green',
      disabled: false,
      width: '10px',
    };
    const view = renderer.create(<ButtonComponent {...props} />);
    expect(view.toJSON()).toMatchSnapshot();
    expect(props.text).toEqual('Button!');
    expect(view.root.props.disabled).toEqual(false);
  });
});
