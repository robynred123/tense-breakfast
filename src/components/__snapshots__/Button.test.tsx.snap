// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> Should render the button component successfully 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1sn7fex-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button!
</button>
`;

exports[`<ButtonComponent /> Should render the button component with gradient 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-4dgfar-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button!
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
